Add section for consensus cell types to methods (#148)
AlexsLemonade/ScPCA-manuscript: content/04.methods.md

For each cell, `CellAssign` calculates a probability of assignment to each cell type in the reference.
The probability matrix and a prediction based on the most probable cell type were added as cell type annotations to the processed `SingleCellExperiment` object output.

#### Assigning consensus cell types

> Cell type labels obtained from `SingleR` and `CellAssign` were then used to assign an ontology-aware consesnsus cell type label. 

Reviewer: ```suggestion
Cell type labels obtained from `SingleR` and `CellAssign` were used to assign an ontology-aware consensus cell type label. 
```

We first assigned each of the cell types present in the `PanglaoDB` [@doi:10.1093/database/baz046] reference used with `CellAssign` to an appropriate Cell Ontology term [@url:https://www.ebi.ac.uk/ols4/ontologies/cl].

> The cell types available in the `BlueprintEncodeData` reference used with `SingleR` already contained Cell Ontology terms, so we did not need to assign these. 

Reviewer: ```suggestion
For cell types available in the `BlueprintEncodeData` reference used with `SingleR`, we used the provided Cell Ontology terms.
```

> The cell types available in the `BlueprintEncodeData` reference used with `SingleR` already contained Cell Ontology terms, so we did not need to assign these. 

Reviewer: ```suggestion
For cell types available in the `BlueprintEncodeData` reference used with `SingleR`, we used the provided Cell Ontology terms.
```

> We then created a reference table that contains all possible combinations of cell types assigned using `SingleR` and `CellAssign` and identifed the latest common ancestor (LCA) [@url:https://rdrr.io/bioc/ontoProc/man/findCommonAncestors.html] between the two cell type terms. 

Reviewer: ```suggestion
We then created a reference table containing all possible combinations of cell types assigned using `SingleR` and `CellAssign` and identified the latest common ancestor (LCA) [@url:https://rdrr.io/bioc/ontoProc/man/findCommonAncestors.html] between the two cell type terms. 
```

> The LCA was then used as the consensus cell type assignment as long as the LCA met the following set of criteria. 

Reviewer: ```suggestion
The LCA was then used as the consensus cell type assignment as long as the LCA met the following set of criteria:
```

If the terms shared more than 1 LCA, then no consensus cell type was assigned. 
The only exception to this rule was if one of the LCA terms corresponded to `hematopoietic precursor cell`. 
If this was the case then the other LCA terms were removed and `hematopoietic precursor cell` was used as the consensus label. 
If the LCA had greater than 170 descendants, then no consensus cell type was assigned, with a few exceptions. 
When the LCA was `neuron`, then `neuron` was used as the consensus cell type label. 
When the LCA was `epithelial cell` and the annotation from `BlueprintEncodeData` was `Epithelial cells`, then `epithelial cell` was used as the consensus label. 

> If the LCA was `bone cell`, `lining cell`, `blood cell`, `progenitor cell`, or `supporting cell`, no consensus label was defined as we thought these terms were not specific enough to define a cell type. 

Reviewer: This section is a bit unclear to me; it isn't always obvious which rule you are applying when. Does it apply to the previous sentence or not?

I think maybe giving these numbered or lettered labels would help, or maybe just combining some sentences. 

Fior exacmple, I think you could say:

"If the terms shared more than one LCA, then no consensus cell type was assigned, unless one of the LCA terms corresponded to `hematopoietic precursor cell`. In that case `hematopoietic precursor cell` was used as the consensus label."

Perhaps:

"LCA terms with a large number of descendants (>170) were not used, except for `neuron` and `epithelial cell`."
Do you need the detail about BluePrintEncodeData here? What is the alternative case? 

And for the last sentence, maybe something like: "We also excluded the following non-specific LCA terms: `bone cell`, `lining cell`, `blood cell`, `progenitor cell`, or `supporting cell`"
